utils.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"net"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Returns the preferedIPProtocol, the dialProtocol, and sets the probeIPProtocolGauge.
+func chooseProtocol(preferredIPProtocol string, transportProtocol string, target string, registry *prometheus.Registry) (string, *net.IPAddr, error) {
+	var fallbackProtocol string
+
+	probeIPProtocolGauge := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "probe_ip_protocol",
+		Help: "Specifies whether probe ip protocl is IP4 or IP6",
+	})
+	registry.MustRegister(probeIPProtocolGauge)
+
+	if preferredIPProtocol == "ip6" || preferredIPProtocol == "" {
+		preferredIPProtocol = "ip6"
+		fallbackProtocol = "ip4"
+	} else {
+		preferredIPProtocol = "ip4"
+		fallbackProtocol = "ip6"
+	}
+
+	if transportProtocol == "icmp4" {
```

Comment:
As we discussed, this function should only be looking at the network protocol - not the transport protocol. Have the transport protocol just be "tcp" without the network protocol mentioned.